service/app.py

```diff
@@ -173,6 +173,33 @@ def get_professor_data():
     # Include the full instructor name in the response
     return jsonify({"professor": full_instructor_name, "courses": course_data})
 
+@app.route('/class', methods=['GET'])
+def get_class_data():
+    class_id = request.args.get('classId') 
+    if not class_id:
+        return jsonify({"error": "Class ID is required"}), 400
+
+    # Fetch class data from the database
+    class_data = ClassData.query.filter_by(class_name=class_id).first()
```

Comment:
I think at this point we should make functions in the database.py file that does the dict stuff since we do it so much but thats for the future